src/lib/deso/transaction.ts

```diff
@@ -335,6 +335,26 @@ export class TransactionMetadataTransferDAOCoin extends BinaryRecord {
   receiverPublicKey: Buffer = Buffer.alloc(0);
 }
 
+export class TransactionMetadataDAOCoinLimitOrder extends BinaryRecord {
+  @Transcode(VarBuffer)
+  buyingDAOCoinCreatorPublicKey: Buffer = Buffer.alloc(0);
+
+  @Transcode(VarBuffer)
+  sellingDAOCoinCreatorPublicKey: Buffer = Buffer.alloc(0);
+
+  @Transcode(FixedBuffer(32))
+  scaledExchangeRateCoinsToSellPerCoinToBuy: Buffer = Buffer.alloc(0);
+
+  @Transcode(FixedBuffer(32))
+  quantityToBuyInBaseUnits: Buffer = Buffer.alloc(0);
+
+  @Transcode(VarBuffer)
+  cancelExistingOrder: boolean = false;
+
+  @Transcode(VarBuffer)
```

Comment:
I think we actually want to create a type for `DeSoInputsByTransactor` and make this `@Transacode(ArrayOf(DeSoInputsByTransactor))`

I think the `DeSoInputsByTransactor` type would look like

```
export class DeSoInputsByTransactor extends BinaryRecord {
  @Transcode(FixedBuffer(33))
  transactorPublicKey: Buffer = Buffer.alloc(0);

  @Transcode(ArrayOf(TransactionInput))
  inputs: TransactionInput[] = [];
}
```

Reply:
Just for my understanding, what are all the reasons we need the to decode the tx metadata in identity? From what I can tell, the primary motivation is to produce a meaningful description of the tx for the user. Anything else?